logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/conversation/MLSWelcomeEventHandler.kt

```diff
@@ -61,22 +64,33 @@ internal class MLSWelcomeEventHandlerImpl(
                 mlsClientProvider.getMLSClient()
             }
             .flatMap { client ->
+                kaliumLogger.d("$TAG: Processing MLS welcome message")
                 wrapMLSRequest {
                     client.processWelcomeMessage(event.message.decodeBase64Bytes())
                 }
             }.flatMap { welcomeBundle ->
                 welcomeBundle.crlNewDistributionPoints?.let {
+                    kaliumLogger.d("$TAG: checking revocation list")
                     checkRevocationList(it)
                 }
+                kaliumLogger.d("$TAG: Marking conversation as established ${welcomeBundle.groupId.obfuscateId()}")
                 markConversationAsEstablished(GroupID(welcomeBundle.groupId))
             }.flatMap {
+                kaliumLogger.d("$TAG: Resolving conversation if one-on-one ${event.conversationId.toLogString()}")
                 resolveConversationIfOneOnOne(event.conversationId)
             }
+            .flatMapLeft {
+                if (it is MLSFailure.ConversationAlreadyExists) {
```

Comment:
Maybe add warning here then? "Discarding welcome since the conversation already exists "